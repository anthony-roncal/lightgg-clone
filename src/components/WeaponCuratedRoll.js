import { React } from 'react';

const WeaponCuratedRoll = ({ socketEntries, BUNGIE_URL, handlePerkHover, handlePerkBlur, handleMouseMove }) => {
    return (
        <>
            <h3 className='section-header'>Curated Roll</h3>
            <p className='curated-roll-info'> Not all curated rolls actually drop in-game.</p>
            <ul className='curated-roll-perks'>
                <li onMouseEnter={handlePerkHover} onMouseLeave={handlePerkBlur} onMouseMove={handleMouseMove}
                    data-perkindex={0} data-testid='perk0'>
                    <img className='curated-roll-perk perk'
                        src={`${BUNGIE_URL}${socketEntries[0].singleInitialItemData.displayProperties.icon}`}
                        alt={`${socketEntries[0].singleInitialItemData.displayProperties.name}`} />
                </li>
                <li onMouseEnter={handlePerkHover} onMouseLeave={handlePerkBlur} onMouseMove={handleMouseMove}
                    data-perkindex={1} data-testid='perk1'>
                    <img className='curated-roll-perk perk'
                        src={`${BUNGIE_URL}${socketEntries[1].singleInitialItemData.displayProperties.icon}`}
                        alt={`${socketEntries[1].singleInitialItemData.displayProperties.name}`} />
                </li>
                <li onMouseEnter={handlePerkHover} onMouseLeave={handlePerkBlur} onMouseMove={handleMouseMove}
                    data-perkindex={2} data-testid='perk2'>
                    <img className='curated-roll-perk perk'
                        src={`${BUNGIE_URL}${socketEntries[2].singleInitialItemData.displayProperties.icon}`}
                        alt={`${socketEntries[2].singleInitialItemData.displayProperties.name}`} />
                </li>
                <li onMouseEnter={handlePerkHover} onMouseLeave={handlePerkBlur} onMouseMove={handleMouseMove}
                    data-perkindex={3} data-testid='perk3'>
                    <img className='curated-roll-perk perk'
                        src={`${BUNGIE_URL}${socketEntries[3].singleInitialItemData.displayProperties.icon}`}
                        alt={`${socketEntries[3].singleInitialItemData.displayProperties.name}`} />
                </li>
                <li onMouseEnter={handlePerkHover} onMouseLeave={handlePerkBlur} onMouseMove={handleMouseMove}
                    data-perkindex={4} data-testid='perk4'>
                    <img className='curated-roll-perk perk'
                        src={`${BUNGIE_URL}${socketEntries[4].singleInitialItemData.displayProperties.icon}`}
                        alt={`${socketEntries[4].singleInitialItemData.displayProperties.name}`} />
                </li>
            </ul>
        </>
    );
}

export default WeaponCuratedRoll;