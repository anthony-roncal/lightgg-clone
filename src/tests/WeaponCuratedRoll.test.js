import React from "react";
import renderer from "react-test-renderer";
import WeaponCuratedRoll from "../components/WeaponCuratedRoll";

const mockData = [
    {
        "singleInitialItemData": {
            "displayProperties": {
                "description": "Rounds fired split into tracking cluster missiles upon detonation.",
                "name": "Wolfpack Rounds",
                "icon": "/common/destiny2_content/icons/9581a5329d9eff75cb770180cc644ced.png",
            }
        }
    },
    {
        "singleInitialItemData": {
            "displayProperties": {
                "description": "This weapon is optimized for an especially explosive payload.\n  •  Greatly increases blast radius\n  •  Slightly decreases handling speed\n  •  Slightly decreases projectile speed",
                "name": "Volatile Launch",
                "icon": "/common/destiny2_content/icons/d255c9140cf640cfc820f995d1c1e96a.png"
            }
        }
    },
    {
        "singleInitialItemData": {
            "displayProperties": {
                "description": "Reduces weight for faster reloads.\n  •  Greatly increases reload speed\n  •  Decreases stability",
                "name": "Alloy Casing",
                "icon": "/common/destiny2_content/icons/e02a542e5162c002112c21b6fdf2e86b.png"
            }
        }
    },
    {
        "singleInitialItemData": {
            "displayProperties": {
                "description": "Gain increased handling and reload speed when standing near allies. Firing this weapon also grants Wolfpack Rounds to nearby allies wielding non-Exotic Rocket Launchers.",
                "name": "Pack Hunter",
                "icon": "/common/destiny2_content/icons/5e8be152a38c4acd61b2a9e56671cc83.png"
            }
        }
    },
    {
        "singleInitialItemData": {
            "displayProperties": {
                "description": "This weapon is especially easy to grip.\n  •  Greatly increases handling speed",
                "name": "Short-Action Stock",
                "icon": "/common/destiny2_content/icons/47f90d3f25a517166933f2b4eb38e9aa.png"
            }
        }
    }
];

const BUNGIE_URL = 'https://www.bungie.net';
const onHoverMock = jest.fn();
const onBlurMock = jest.fn();
const onMouseMoveMock = jest.fn();


describe("Weapon Curated Roll component", () => {
    it("renders correct elements and data", () => {
        const tree = renderer.create(
            <WeaponCuratedRoll
                socketEntries={mockData}
                BUNGIE_URL={BUNGIE_URL}
                handlePerkHover={onHoverMock}
                handlePerkBlur={onBlurMock}
                handleMouseMove={onMouseMoveMock}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
