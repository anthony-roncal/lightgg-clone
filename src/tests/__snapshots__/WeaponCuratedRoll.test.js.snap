// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weapon Curated Roll component renders correct elements and data 1`] = `
Array [
  <h3
    className="section-header"
  >
    Curated Roll
  </h3>,
  <p
    className="curated-roll-info"
  >
     Not all curated rolls actually drop in-game.
  </p>,
  <ul
    className="curated-roll-perks"
  >
    <li
      data-perkindex={0}
      data-testid="perk0"
      onMouseEnter={[MockFunction]}
      onMouseLeave={[MockFunction]}
      onMouseMove={[MockFunction]}
    >
      <img
        alt="Wolfpack Rounds"
        className="curated-roll-perk perk"
        src="https://www.bungie.net/common/destiny2_content/icons/9581a5329d9eff75cb770180cc644ced.png"
      />
    </li>
    <li
      data-perkindex={1}
      data-testid="perk1"
      onMouseEnter={[MockFunction]}
      onMouseLeave={[MockFunction]}
      onMouseMove={[MockFunction]}
    >
      <img
        alt="Volatile Launch"
        className="curated-roll-perk perk"
        src="https://www.bungie.net/common/destiny2_content/icons/d255c9140cf640cfc820f995d1c1e96a.png"
      />
    </li>
    <li
      data-perkindex={2}
      data-testid="perk2"
      onMouseEnter={[MockFunction]}
      onMouseLeave={[MockFunction]}
      onMouseMove={[MockFunction]}
    >
      <img
        alt="Alloy Casing"
        className="curated-roll-perk perk"
        src="https://www.bungie.net/common/destiny2_content/icons/e02a542e5162c002112c21b6fdf2e86b.png"
      />
    </li>
    <li
      data-perkindex={3}
      data-testid="perk3"
      onMouseEnter={[MockFunction]}
      onMouseLeave={[MockFunction]}
      onMouseMove={[MockFunction]}
    >
      <img
        alt="Pack Hunter"
        className="curated-roll-perk perk"
        src="https://www.bungie.net/common/destiny2_content/icons/5e8be152a38c4acd61b2a9e56671cc83.png"
      />
    </li>
    <li
      data-perkindex={4}
      data-testid="perk4"
      onMouseEnter={[MockFunction]}
      onMouseLeave={[MockFunction]}
      onMouseMove={[MockFunction]}
    >
      <img
        alt="Short-Action Stock"
        className="curated-roll-perk perk"
        src="https://www.bungie.net/common/destiny2_content/icons/47f90d3f25a517166933f2b4eb38e9aa.png"
      />
    </li>
  </ul>,
]
`;
